### ISSUE TITLE
Bug: Regression: can't edit regexes of whitespace chars

### ISSUE THREAD

**Issue by todated** [platers/obsidian-linter#1591]
- [x] I have verified that I am on the latest version of the Linter

issue started with v1.33.0

## Describe the Bug

regexes that find only whitespace chars cannot (any longer) be edited in the new editor interface

the validation in the settings dialog complains "Invalid regex" or "Required" 
(seems to be wrongly assuming the input field is empty)

the "regex to replace" field doesn't complain if the field contains only whitespace chars; only the "regex to find" field complains;

## How to Reproduce

`Settings > Linter > Custom > Custom regex replacement > +`

Add new regex replacement

create a 'regex to find' such as:

` +$`

try to 'Save'

Example to reproduce issue with

my `data.json` has custom regexes such as these:
(which i created using a previous version of this extension, without needing to manually edit `data.json`)

```json

"customRegexes": [
    {
      "label": "tab \\t => 4 spaces",
      "find": "\t",
      "flags": "gm",
      "replace": "    ",
      "enabled": true
    },
    {
      "label": "&nbsp; => ' '",
      "find": " ",
      "flags": "gm",
      "replace": " ",
      "enabled": true
    },
    {
      "label": "trailing spaces",
      "find": " +$",
      "flags": "gm",
      "replace": "",
      "enabled": true
    }
]

```

## Expected Behavior

i should be able to edit/save a 'regex to find' that has non-zero length, even if it's whitespace chars, just as was already possible in versions prior to the latest;

## Screenshots

<img width="638" height="552" alt="Image" src="https://github.com/user-attachments/assets/59f903ed-159a-4876-ba74-acaf0492654b" />

## Work-arounds

the regexes still work correctly, but if i want to make any changes to them i have now edit this file by hand;
`.../.obsidian/plugins/obsidian-linter/data.json`

## Device

- [x] Desktop Linux
- [ ] Mobile

**Comment by todated** [comment]
PS, i'm aware that the linter has many of it's own "Spacing" rules built-in
(and i use a number of them);

however, if those rules don't *quite* cover some of my use cases, and i have to create some custom regexes anyway, then i'd prefer to just be able to see all of my (whitespace-handling) regexes in the custom list;

**Comment by pjkaufman** [comment]
This makes sense as a bug. It was just an oversight when the settings got converted.
